universe/supplyverifier/manager.go

```diff
@@ -0,0 +1,567 @@
+package supplyverifier
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/proof"
+	"github.com/lightninglabs/taproot-assets/tapgarden"
+	"github.com/lightninglabs/taproot-assets/universe/supplycommit"
+	"github.com/lightningnetwork/lnd/msgmux"
+	"github.com/lightningnetwork/lnd/protofsm"
+)
+
+const (
+	// DefaultTimeout is the context guard default timeout.
+	DefaultTimeout = 30 * time.Second
+)
+
+// DaemonAdapters is a wrapper around the protofsm.DaemonAdapters interface
+// with the addition of Start and Stop methods.
+type DaemonAdapters interface {
+	protofsm.DaemonAdapters
+
+	// Start starts the daemon adapters handler service.
+	Start() error
+
+	// Stop stops the daemon adapters handler service.
+	Stop() error
+}
+
+// StateMachineStore is an interface that allows the state machine to persist
+// its state across restarts. This is used to track the state of the state
+// machine for supply verification.
+type StateMachineStore interface {
+	// CommitState is used to commit the state of the state machine to disk.
+	CommitState(context.Context, asset.Specifier, State) error
+
+	// FetchState attempts to fetch the state of the state machine for the
+	// target asset specifier. If the state machine doesn't exist, then a
+	// default state will be returned.
+	FetchState(context.Context, asset.Specifier) (State, error)
+}
+
+// IssuanceSubscriptions allows verifier state machines to subscribe to
+// asset group issuance events.
+type IssuanceSubscriptions interface {
+	// RegisterSubscriber registers an event receiver to receive future
```

Comment:
Do we have a way to handle dropped events here and/or restarts? 

So an issuance event happens (we detect this in the normal universe syncer), then we want to notify this sub-system. What happens if we send that event, write the new issuance to disk, then crash? You can reverse the order of send vs write as well.